custom/icds_reports/management/commands/fill_missing_gm_forms.py

```diff
@@ -0,0 +1,38 @@
+from django.core.management import BaseCommand
+from datetime import datetime
+
+from custom.icds_reports.const import (
+    DASHBOARD_DOMAIN
+)
+
+from corehq.apps.locations.models import SQLLocation
+from custom.icds_reports.tasks import (
+    _get_monthly_dates,
+    icds_state_aggregation_task
+)
+
+
+class Command(BaseCommand):
+    help = "Fills the Missing GM Forms"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'date', type=str, help='Date of the month whose stunting and wasting data is missing yyyy-mm-dd')
+
+    def handle(self, *args, **kwargs):
+        date = kwargs['date']
+        date = datetime.strptime(date, '%Y-%m-%d')
+        self.run_task(date)
+
+    def run_task(self, date):
+        initial_date = datetime(2017, 4, 1, 0, 0)
+        # get the number of month intervals between the dates
+        intervals = date.month - initial_date.month + 12 * (date.year - initial_date.year) + 1
+        monthly_dates = _get_monthly_dates(date, total_intervals=intervals)
+
+        state_ids = list(SQLLocation.objects.filter(domain=DASHBOARD_DOMAIN,
+                                                    location_type__name='state').values_list('location_id', flat=True))
+        for monthly_date in monthly_dates:
+            for state_id in state_ids:
+                icds_state_aggregation_task(state_id=state_id, date=monthly_date,
```

Comment:
I dont think this is how you want to do this. Based on the speed that builds, it will take at least 48 hours to build all these tables, while they are building the agg will be slowed down, and as soon as the agg runs it will delete, whatever you have managed to build up to that point in its automatic cleanup. If the issue here is just eligibility, is it possible to use data that is already in `child_health_monthly`? It looks like most of these values are recorded in the table already, without the eligibility checks. You should be able to just copy those if the case meets the new eligibility, rather than reaggregating from scratch, and that would be much more performant. I think it could actually make most of this unnecessary.

Reply:
I agree to this point to some extent but this field
```sql
height_measured_in_month = CASE
         WHEN (gm.height_child_last_recorded >= '%(start_date)s' AND gm.height_child_last_recorded <'%(end_date)s') AND (valid_in_month=1 AND age_tranche::Integer <= 60) THEN 1
         ELSE
           0
         END,
```
is completely dependent on the gm table

Reply:
why is that? it appears the height part of that from the gm table is exactly replicated in `recorded_height` in child health monthly. so that expression could be
```
height_measured_in_month = CASE
         WHEN (recorded_recorded is not null) AND (valid_in_month=1 AND age_tranche::Integer <= 60) THEN 1
         ELSE
           0
         END,
```

Reply:
Updated!!